### DISCUSSION TITLE
SiftGPU not fully supported

### DISCUSSION THREAD

**Discussion by AdamGoodApp** [NVlabs/instant-ngp#225] [Q&A] [answered]
Running the colmap2nerf.py to setup video for custom training.

`python3 ../../../scripts/colmap2nerf.py --video_in testvid.mp4 --video_fps 2 --run_colmap --aabb_scale 16`

```
Shader not supported by your hardware!
ERROR: SiftGPU not fully supported.
```

testbed works fine.
CUDA, FFMPEG, Colmap installed ok.

Running in WSL2, Nvidia RTX 3090 GPU.

**Comment by Tom94** [comment, marked as answer]
Hi there, this is unfortunately a limitation of the underlying COLMAP call, so I'd forward you to their issue tracker over here: https://github.com/colmap/colmap

Alternatively, you could try running the colmap reconstruction natively on Window, only running __instant-ngp__ on WSL2 after the fact.

**Reply by nowhereman1999** [reply]
> Hi there, this is unfortunately a limitation of the underlying COLMAP call, so I'd forward you to their issue tracker over here: https://github.com/colmap/colmap
> 
> Alternatively, you could try running the colmap reconstruction natively on Window, only running **instant-ngp** on WSL2 after the fact.

Actually it can be solved by build the colmap from source with cuda.This is how I solved it on wsl2
